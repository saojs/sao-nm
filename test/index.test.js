const path = require('path')
const sao = require('sao')

const generator = path.join(__dirname, '..')

const getPkg = (pkg, fields) => {
  pkg = JSON.parse(pkg)
  return fields.reduce((res, curr) => {
    res[curr] = pkg[curr]
    return res
  }, {})
}

test('use defaults', async () => {
  const helper = await sao.mock({ generator })
  expect(helper.fileList).toMatchSnapshot()
})

test('add unit test', async () => {
  const helper = await sao.mock({ generator }, {
    unitTest: true
  })

  expect(helper.fileList).toMatchSnapshot('files')
  expect(
    getPkg(await helper.readFile('package.json'), ['scripts', 'devDependencies'])
  ).toMatchSnapshot('package.json')
})

test('add coverage', async () => {
  const helper = await sao.mock({ generator }, {
    unitTest: true,
    coverage: true
  })

  expect(helper.fileList).toMatchSnapshot('files')
  expect(await helper.readFile('circle.yml')).toMatchSnapshot('circle.yml')
})

test('add cli', async () => {
  const helper = await sao.mock({ generator }, {
    cli: true
  })
  expect(helper.fileList).toMatchSnapshot('files')
  expect(
    getPkg(await helper.readFile('package.json'), ['bin', 'dependencies'])
  ).toMatchSnapshot('package.json')
})
