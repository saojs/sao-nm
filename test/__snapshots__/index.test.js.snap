// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add cli: files 1`] = `
Array [
  ".editorconfig",
  ".gitattributes",
  ".gitignore",
  ".prettierrc",
  "LICENSE",
  "README.md",
  "circle.yml",
  "cli.js",
  "index.js",
  "package.json",
]
`;

exports[`add cli: package.json 1`] = `
Object {
  "bin": "cli.js",
  "dependencies": Object {
    "cac": "^6.0.0",
  },
}
`;

exports[`add coverage: circle.yml 1`] = `
"version: 2
jobs:
  build:
    docker:
      - image: circleci/node:latest
    branches:
      ignore:
        - gh-pages # list of branches to ignore
        - /release\\\\/.*/ # or ignore regexes
    steps:
      - checkout
      - restore_cache:
          key: dependency-cache-{{ checksum \\"yarn.lock\\" }}
      - run:
          name: install dependences
          command: yarn
      - save_cache:
          key: dependency-cache-{{ checksum \\"yarn.lock\\" }}
          paths:
            - ./node_modules
      - run:
          name: test
          command: yarn test:cov
      - run:
          name: upload coverage
          command: bash <(curl -s https://codecov.io/bash)
"
`;

exports[`add coverage: files 1`] = `
Array [
  ".editorconfig",
  ".gitattributes",
  ".gitignore",
  ".prettierrc",
  "LICENSE",
  "README.md",
  "circle.yml",
  "index.js",
  "package.json",
  "test/index.test.js",
]
`;

exports[`add unit test: files 1`] = `
Array [
  ".editorconfig",
  ".gitattributes",
  ".gitignore",
  ".prettierrc",
  "LICENSE",
  "README.md",
  "circle.yml",
  "index.js",
  "package.json",
  "test/index.test.js",
]
`;

exports[`add unit test: package.json 1`] = `
Object {
  "devDependencies": Object {
    "eslint-config-prettier": "^3.3.0",
    "eslint-config-rem": "^4.0.0",
    "eslint-plugin-prettier": "^3.0.0",
    "husky": "^1.0.0-rc.13",
    "jest": "^23.6.0",
    "lint-staged": "^7.2.0",
    "prettier": "^1.15.2",
    "xo": "^0.23.0",
  },
  "scripts": Object {
    "lint": "xo",
    "test": "npm run lint && jest",
  },
}
`;

exports[`use defaults 1`] = `
Array [
  ".editorconfig",
  ".gitattributes",
  ".gitignore",
  ".prettierrc",
  "LICENSE",
  "README.md",
  "circle.yml",
  "index.js",
  "package.json",
]
`;
